src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,57 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
 
+    private final Map<Class<?>, Class<?>> interfaceImplementations =
+            Map.of(FileReaderService.class, FileReaderServiceImpl.class,
+                    ProductParser.class, ProductParserImpl.class,
+                    ProductService.class, ProductServiceImpl.class);
+
+    private final Map<Class<?>, Object> instances = new HashMap<>();
+
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> implementationClass = findImplementation(interfaceClazz);
+        if (implementationClass.isAnnotationPresent(Component.class)) {
+            return createNewInstance(implementationClass);
+        } else {
+            throw new RuntimeException("Class " + implementationClass.getName()
+                    + " is not annotated with @Component");
+        }
+    }
```

Comment:
You forget to implement logic about @Inject. For now, this method doesn't initialize fields that should be initialize